src/network/networkd-nexthop.c

```diff
@@ -406,9 +406,15 @@ int manager_rtnl_process_nexthop(sd_netlink *rtnl, sd_netlink_message *message,
         }
 
         r = sd_netlink_message_read_u32(message, NHA_ID, &tmp->id);
-        if (r < 0 && r != -ENODATA) {
+        if (r == -ENODATA) {
+                log_link_warning_errno(link, r, "rtnl: received nexthop message without NHA_ID attribute, ignoring: %m");
+                return 0;
+        } else if (r < 0) {
                 log_link_warning_errno(link, r, "rtnl: could not get NHA_ID attribute, ignoring: %m");
                 return 0;
+        } else if (tmp->id == 0) {
+                log_link_warning_errno(link, r, "rtnl: received nexthop message with invalid nexthop ID, ignoring: %m");
```

Comment:
second argument should probably be explicitly set to zero, given that r doesn#t really carry an error here, but doesn't matter, since it's gonna be 0 here in that case anyway. but still looks like a bug, even though it isn't